sql/create_bldgs_table.sql

```diff
@@ -35,12 +35,43 @@ rentstab as (
     coalesce(uc2019, 0) - coalesce(unitsstab2007, 0) rsdiff
   from rentstab_summary
   left join rentstab_v2 using(ucbbl)
+),
+
+complaints as (
+  select 
+    bbl, 
+    sum(countcomplaints) as totalcomplaints, 
+    json_agg(
+        json_build_object(complainttype,countcomplaints) 
+        order by countcomplaints desc) complaintsbytype
```

Comment:
Soooo, I was on the fence regarding exactly what structure I wanted to use in building these json outputs. I originally structured these outputs to have separate key/value pairs for both the complainttype and the count, so outputs in the SQL table would look like this:
```
[{type: "HEAT/HOT WATER", count: 4}, {type: "PESTS", count: 2}]
```
This seemed to be the standard way to set up the json object given that both the type and count fields are considered "data." 

However, I also took into account that whatever raw format the json object is in is what users see when they click the "Export Data" button and download the raw data for a given portfolio. Given that, I thought it would be a lot more digestible and readable for a user to see this json object in a more condensed form, with the complainttypes as keys and the counts as values:
```
[{"HEAT/HOT WATER": 4}, {"PESTS": 2}]
```
This is what I landed on, but I just wanted to check whether this would make sense in the long run. I didn't see any major issue in trying to pull these values into our front-end, but lmk!

Reply:
Hmm, so there are a few advantages I see with the "standard" (first) approach you described:

1. It's really straightforward to define this kind of type in TypeScript.  It's also a natural fit for an algebraic type, so if for whatever reason you wanted to add extra fields that are specific to particular types of repairs, it would be really easy.

2. In general, it's easy to add extra information to each blob in a forwards-compatible way. For instance, if for some reason you wanted to mention, say, the date of the most recent complaint in a particular category, you could just add a `latestComplaintDate` field without any issue. Out-of-date front-ends would still be fine with parsing the new object. This is in contrast to the second format you provided, where adding any additional information would either require the value of each object to change from a number to an object, or something else that out-of-date front-ends might explode on.

That said, readability of the export data feed also sounds important, so I'm not really sure which way to go here.  I guess I might lean towards an internal representation that's more verbose (the first option) but have the exported CSV perhaps contain the more condensed form, if that's even a possibility, but I forget enough about the actual implementation right now that I'm not sure if it is!

Reply:
Ok, given your two comments, I decided to refactor the SQL to spit out the "standard" json format, and will work to make sure the output in the data export is readable (I think we have some mechanism to do this for other fields too).